EIPS/eip-4469.md

```diff
@@ -0,0 +1,55 @@
+---
+eip: 4469
+title: Wrapped metaverse token
+description: Defines a standard interface for viewing the addresses and balances of wrapped metaverse tokens.
+author: Hyungsuk Kang (@hskang9)
+discussions-to: https://ethereum-magicians.org/t/eip-4469-wrapping-erc1155-into-erc20s/7516
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-11-21
+---
+
+## Abstract
+
+Metaverses(e.g. Decentraland, SandBox, etc) are now emerging in Ethereum ecosystem, and the standard of the tokens in each metaverse are all aggregated into ERC-1155 for efficiency.
+Invented by Witek Radomski from Enjin, this has made many metaverse/game project to accomodate their assets to provide ownerships. 
+However, current ERC-1155 has been sacrificing its interoperability to other universes and current defi ecosystem due to different data structure and mechanics. A certain contract to bridge them into prepared ERC-20 could be possible, but this just can create duplicate datas in the global state if source is not uniform. Due to this, I propose the standard wrapped metaverse token contract to wrap metaverse assets into the most widely used ERC-20 tokens. 
+
+This EIP introduces a simple interface for exposing the addresses of all wrapped ERC-1155 tokens in metaverses, as well as querying wrapped assets' balances.
+
```

Comment:
Hi! I've been looking over your ERC, and have a few suggestions.

For starters, I'm looking over the description, and while I assume your personal usecase specifically involves metaverses, it doesn't seem _technically_ integral to the ERC. If I had to summarize the technical aspect of the ERC, it looks to me (please correct me if I'm wrong) like a standard for wrapping and unwrapping ERC1155 assets in ERC20s. There's nothing wrong with that! But you might want to think about making that the focus here, and editing out the parts about metaverses since they don't seem to be a technical part of the standard.

Reply:
I agree. This has to be more generic for providing compatibility between erc20 and erc721/1155.